nix/solidity-package.sh

```diff
@@ -25,7 +19,26 @@ if [ "$doCheck" == 1 ]; then
   dapp2-test-hevm
 fi
 
-mkdir -p $out/dapp/$name
-cp -r $src $out/dapp/$name/src
-cp -r lib $out/dapp/$name/lib
-cp -r out $out/dapp/$name/out
+if [ "$extract" == 1 ]; then
+  mapfile -t contracts < <(<"$DAPP_JSON" jq '.contracts|keys[]' -r | sort -u -t: -k2 | sort)
+  data=$(<"$DAPP_JSON" jq '.contracts' -r)
+  count=1
+  total=${#contracts[@]}
+  for path in "${contracts[@]}"; do
+    info -ne "Extracting build data... [$count/$total]\\r"
+    ((count++))
+    name="${path#*:}"
```

Comment:
This `name` might be overwriting `name` set by the derivation name.

Reply:
Changing it just in case.